skyfield/timelib.py

```diff
@@ -210,19 +210,21 @@ class Time(object):
     eps_correction = 0.0
 
     def __init__(self, ts, tt):
-        self.tt = tt
+        self.tt = _to_array(tt)
         self.ts = ts
         self.shape = getattr(tt, 'shape', ())
 
     def __len__(self):
         return self.shape[0]
 
     def __repr__(self):
-
-        if getattr(self.tt, 'shape', ()):
+        if self.tt.size == 0:
```

Comment:
I'm not seeing how this avoids raising an exception immediately if `self.tt` is a normal Python float?

Reply:
This doesn't avoid raising the exception, instead it depends on `self.tt` never being a python float, which is why I changed line 213 to use `_to_array`. If users never instantiate a Time object directly, then `self.tt` being a python float is very unlikely, and maybe this is fine as is? Otherwise it should change to be able to handle that.

Reply:
Understood. Try removing the `_to_array()` up above, and then pivoting this to be `getattr(self.tt, 'size', 1)` so that this class remains, for the moment at least, agnostic about whether it's been given a Python float or not. Thanks!